src/executor.rs

```diff
@@ -166,6 +188,14 @@ fn invoke_dynamic(
                 (syscall_handler as *mut StarknetSyscallHandlerCallbacks<_>)
                     .to_bytes(&mut invoke_data)?;
             }
+            CoreTypeConcrete::BuiltinCosts(_) => {
+                // This builtin should never be an argument but just in case.
+                if let Some(builtin_costs) = builtin_costs {
+                    builtin_costs.to_bytes(&mut invoke_data)?;
+                } else {
+                    (builtin_costs_array.as_ptr()).to_bytes(&mut invoke_data)?;
+                }
+            }
```

Comment:
This match branch will make the changes below (line 282 after changes) unreachable.

Reply:
i dont see it, can you explain in more detail?

Reply:
Nevermind I'm blind.